### PR TITLE
Add support for java.time.ZoneId

### PR DESCRIPTION
**Pull request by dragisak**

### CODE REVIEW

File: play-json/src/main/scala/play/api/libs/json/Reads.scala [hunk 1]
```diff
@@ -604,6 +604,23 @@ trait DefaultReads extends LowPriorityDefaultReads {
     instantReads(DateTimeFormatter.ISO_DATE_TIME)
 
   /**
+   * Reads for the `java.time.ZoneId` type.
+   */
+  implicit val ZoneIdReads :Reads[ZoneId]= new Reads[ZoneId] {
```
Comment (cchantep) [inline play-json/src/main/scala/play/api/libs/json/Reads.scala]: `: Reads[..] = Reads[..] { .. }`

File: play-json/src/main/scala/play/api/libs/json/Reads.scala [hunk 2]
```diff
@@ -604,6 +604,23 @@ trait DefaultReads extends LowPriorityDefaultReads {
...
+    override def reads(json: JsValue): JsResult[ZoneId] = json match {
+      case JsString(s) => try {
+        JsSuccess(ZoneId.of(s))
+      } catch {
+        case _: DateTimeException =>
+          JsError(Seq(JsPath() ->
```
Comment (cchantep) [inline play-json/src/main/scala/play/api/libs/json/Reads.scala]: `JsError(JsValidationError(..))`

File: play-json/src/main/scala/play/api/libs/json/Reads.scala [hunk 3]
```diff
@@ -604,6 +604,23 @@ trait DefaultReads extends LowPriorityDefaultReads {
...
+            Seq(JsonValidationError("error.expected.timezone", s))))
+      }
+      case _ =>  JsError(Seq(JsPath() ->
```
Comment (cchantep) [inline play-json/src/main/scala/play/api/libs/json/Reads.scala]: Same

File: play-json/src/main/scala/play/api/libs/json/Writes.scala [hunk 4]
```diff
@@ -350,6 +350,13 @@ trait DefaultWrites extends LowPriorityWrites {
   }
 
   /**
+   * Serializer for `java.time.ZoneId` as JSON string.
+   */
+  implicit val ZoneIdWrites = new Writes[ZoneId] {
```
Comment (cchantep) [inline play-json/src/main/scala/play/api/libs/json/Writes.scala]: `: Writes[..] = Writes[..] { .. }`

File: play-json/src/main/scala/play/api/libs/json/Reads.scala [hunk 5]
```diff
@@ -604,6 +604,18 @@ trait DefaultReads extends LowPriorityDefaultReads {
     instantReads(DateTimeFormatter.ISO_DATE_TIME)
 
   /**
+   * Reads for the `java.time.ZoneId` type.
+   */
+  implicit val ZoneIdReads: Reads[ZoneId] = Reads[ZoneId] {
+    case JsString(s) => try {
+        JsSuccess(ZoneId.of(s))
```
Comment (cchantep) [inline play-json/src/main/scala/play/api/libs/json/Reads.scala]: Indent issue

File: play-json/src/main/scala/play/api/libs/json/Reads.scala [hunk 6]
```diff
@@ -604,6 +604,18 @@ trait DefaultReads extends LowPriorityDefaultReads {
...
+      } catch {
+        case _: DateTimeException => JsError(JsonValidationError("error.expected.timezone", s))
+      }
+    case _ => JsError(JsonValidationError("error.expected.jsstring"))
```
Comment (cchantep) [inline play-json/src/main/scala/play/api/libs/json/Reads.scala]: Blank line before case

File: play-json/src/main/scala/play/api/libs/json/Writes.scala [hunk 7]
```diff
@@ -350,6 +350,11 @@ trait DefaultWrites extends LowPriorityWrites {
   }
 
   /**
+   * Serializer for `java.time.ZoneId` as JSON string.
+   */
+  implicit val ZoneIdWrites = Writes[ZoneId](zone => JsString(zone.getId))
```
Comment (cchantep) [inline play-json/src/main/scala/play/api/libs/json/Writes.scala]: Explicit type (recommended for implicit)

File: play-json/src/main/scala/play/api/libs/json/Reads.scala (hunk 6 above)
Comment (dragisak) [inline play-json/src/main/scala/play/api/libs/json/Reads.scala]: Are you asking me to put a blank like before `case _ => JsError...` ? I can't find an example in the rest of the code. 

Would this be the right formatting ?
```scala
 implicit val ZoneIdReads: Reads[ZoneId] = Reads[ZoneId] {
    case JsString(s) => try {
      JsSuccess(ZoneId.of(s))
    } catch {
      case _: DateTimeException => JsError(JsonValidationError("error.expected.timezone", s))
    }
      
    case _ => JsError(JsonValidationError("error.expected.jsstring"))
  }
```

Comment (cchantep) [inline play-json/src/main/scala/play/api/libs/json/Reads.scala]: yes

**Comment by cchantep** [comment]
@gmethvin should be ok (cannot approve/merge)

**Pull request merged**